### ISSUE TITLE
Changing thumbnail of the channel in the staging tree also affects the main tree channel node

### ISSUE THREAD

**Issue by lyw07** [learningequality/ricecooker#241]
* ricecooker version: 0.6.38
* Python version: 3.7
* Operating System: MacOSX

### Description

I removed the thumbnail in the channel node when running sushi chef with `--stage` flag, but this also removed the main channel's thumbnail

### What I Did

```
Paste the command(s) you ran and the output.
If there was a crash, please include the traceback here.
```
1. An existing channel with a thumbnail
2. In the sushi chef code, remove the thumbnail field from the channel node
3. run the sushi chef with `--stage` flag
4. check that the channel's thumbnail in studio production has been removed

**Comment by kollivier** [comment]
This may actually be a Studio issue. The issue is that while Channels have a staging tree, the Channel model itself has no concept of versioning or staging updates. So any changes to the Channel metadata / information will take effect immediately. We may be able to workaround this by checking if the channel exists when using `--stage` and not passing any channel information in that case, but that creates a not fully updated channel and could lead to confusion where it has different thumbnails in different views, etc.

Pinging @jayoshih for ideas on how we could fix this. Maybe it's really time to put channel versioning on the roadmap?

**Comment by jayoshih** [comment]
Ah yes, this is definitely a Studio issue. Unfortunately, there isn't really a way to stage thumbnails at the Channel model level. There are a few approaches that we could do to fix this:
1. Make channel thumbnails reference the tree roots instead, but this will require a migration
2. Add a new staged_thumbnail field
3. A combination of the two where we set the root topic's thumbnail of the staging_tree and make that the channel thumbnail on activation

I'm leaning towards 3 as it leverages the existing model schema without introducing the need for a big migration

**Comment by kollivier** [comment]
@jayoshih This isn't a problem specifically for thumbnails, but all Channel information. If an update changes language, description, or any other information about a channel, we will have the same problem where the published tree is the old tree, but the information matches the new run.

**Comment by jayoshih** [comment]
We could probably use the same approach for those as well- just make the root of the staging_tree the new channel fields (either using the fields that already exist or setting the extra_fields field

**Comment by kollivier** [comment]
Okay, if we could just always defer to the fields in main_tree rather than fields on the Channel model itself, that would be a simple solution! We could in the short term start by just changing the code to access the tree's fields rather than accessing the channel fields directly, then do the migration removing the fields when it would be convenient.

**Comment by jayoshih** [comment]
I'm not sure how easy that would be considering all of the existing data, but it does seem more doable for staging_trees at least

**Comment by kollivier** [comment]
Yeah, as a start we'd only switch to using the staging tree for review sections in the UX, then update the actual channel fields at the end of the publish operation.

**Comment by lyw07** [comment]
thank you both! should I move this issue to studio repo?

**Comment by jayoshih** [comment]
@lyw07 Yes, that would be great! Thanks for opening this issue
